comms/dht/tests/dht.rs

```diff
@@ -536,6 +540,85 @@ async fn dht_propagate_dedup() {
     assert_eq!(count_messages_received(&received, &[&node_C_id]), 1);
 }
 
+#[tokio_macros::test]
+#[allow(non_snake_case)]
+async fn dht_repropagate() {
+    // Node C knows no one
+    let mut node_C = make_node(PeerFeatures::COMMUNICATION_NODE, []).await;
+    // Node B knows about Node C
+    let mut node_B = make_node(PeerFeatures::COMMUNICATION_NODE, [node_C.to_peer()]).await;
+    // Node A knows about Node B and C
+    let mut node_A = make_node(PeerFeatures::COMMUNICATION_NODE, [node_B.to_peer(), node_C.to_peer()]).await;
+    node_A.comms.peer_manager().add_peer(node_C.to_peer()).await.unwrap();
+    log::info!(
+        "NodeA = {}, NodeB = {}, Node C = {}",
+        node_A.node_identity().node_id().short_str(),
+        node_B.node_identity().node_id().short_str(),
+        node_C.node_identity().node_id().short_str(),
+    );
+
+    // Connect the peers that should be connected
+    async fn connect_nodes(node1: &mut TestNode, node2: &mut TestNode) {
+        node1
+            .comms
+            .connectivity()
+            .dial_peer(node2.node_identity().node_id().clone())
+            .await
+            .unwrap();
+    }
+    // Pre-connect nodes, this helps message passing be more deterministic
+    connect_nodes(&mut node_A, &mut node_B).await;
+    connect_nodes(&mut node_B, &mut node_C).await;
+    connect_nodes(&mut node_C, &mut node_A).await;
+
+    #[derive(Clone, PartialEq, ::prost::Message)]
+    struct Person {
+        #[prost(string, tag = "1")]
+        name: String,
+        #[prost(uint32, tag = "2")]
+        age: u32,
+    }
+
+    let out_msg = OutboundDomainMessage::new(123, Person {
+        name: "John Conway".into(),
+        age: 82,
+    });
+    node_A
+        .dht
+        .outbound_requester()
+        .propagate(
+            NodeDestination::Unknown,
+            OutboundEncryption::ClearText,
+            vec![],
+            out_msg.clone(),
+        )
+        .await
+        .unwrap();
+
+    let msg = node_B
+        .next_inbound_message(Duration::from_secs(10))
+        .await
+        .expect("Node expected an inbound message but it never arrived");
+    log::info!("Received message {}", msg.tag);
+
+    node_C
+        .dht
+        .outbound_requester()
+        .propagate(NodeDestination::Unknown, OutboundEncryption::ClearText, vec![], out_msg)
+        .await
+        .unwrap();
```

Comment:
None should receive?

Reply:
I didn't have time to complete this test, this can either go in or I will complete this test on Sunday.

Reply:
Thanks for clearing that up, at least I am on the same page then